presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoContainerBasicQueries.java

```diff
@@ -0,0 +1,131 @@
+    private static final String BASE_DIR = System.getProperty("user.dir");
+    private static final Network network = Network.newNetwork();
+    private GenericContainer<?> coordinator;
+    private GenericContainer<?> worker;
+
+    @BeforeClass
+    public void setUp()
+            throws InterruptedException
+    {
+        coordinator = new GenericContainer<>(PRESTO_COORDINATOR_IMAGE)
+                .withExposedPorts(8081)
+                .withNetwork(network).withNetworkAliases("presto-coordinator")
+                .withFileSystemBind(BASE_DIR + "/testcontainers/coordinator/etc", "/opt/presto-server/etc", BindMode.READ_WRITE)
+                .withFileSystemBind(BASE_DIR + "/testcontainers/coordinator/entrypoint.sh", "/opt/entrypoint.sh", BindMode.READ_ONLY)
+                .waitingFor(Wait.forLogMessage(".*======== SERVER STARTED ========.*", 1))
+                .withStartupTimeout(Duration.ofSeconds(120));
+
+        worker = new GenericContainer<>(PRESTO_WORKER_IMAGE)
+                .withExposedPorts(7777)
+                .withNetwork(network).withNetworkAliases("presto-worker")
+                .withFileSystemBind(BASE_DIR + "/testcontainers/nativeworker/velox-etc", "/opt/presto-server/etc", BindMode.READ_ONLY)
+                .withFileSystemBind(BASE_DIR + "/testcontainers/nativeworker/entrypoint.sh", "/opt/entrypoint.sh", BindMode.READ_ONLY)
+                .waitingFor(Wait.forLogMessage(".*Announcement succeeded: HTTP 202.*", 1));
+
+        coordinator.start();
+        worker.start();
+
+        // Wait for worker to announce itself.
+        TimeUnit.SECONDS.sleep(5);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        coordinator.stop();
+        worker.stop();
+    }
+
+    private Container.ExecResult executeQuery(String sql)
+            throws IOException, InterruptedException
+    {
+        // Command to run inside the coordinator container using the presto-cli.
+        String[] command = {
+                "/opt/presto-cli",
+                "--server",
+                "presto-coordinator:8081",
+                "--catalog",
+                "tpch",
+                "--schema",
+                "tiny",
+                "--execute",
+                sql
+        };
+
+        Container.ExecResult execResult = coordinator.execInContainer(command);
+        if (execResult.getExitCode() != 0) {
+            String errorDetails = "Stdout: " + execResult.getStdout() + "\nStderr: " + execResult.getStderr();
+            fail("Presto CLI exited with error code: " + execResult.getExitCode() + "\n" + errorDetails);
+        }
+        return execResult;
+    }
+
+    @Test
+    public void testBasics()
+            throws IOException, InterruptedException
+    {
+        String selectRuntimeNodes = "select * from system.runtime.nodes";
+        Container.ExecResult nodes = executeQuery(selectRuntimeNodes);
```

Comment:
Would be great to abstract a method that can take as input a query and expected results for all tests below.